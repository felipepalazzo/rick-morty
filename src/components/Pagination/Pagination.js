import React from 'react'
import PropTypes from 'prop-types'
import './Pagination.css'

export default function Pagination({ current, total, onClick }) {
	let pages = []
	for (let i = 1; i <= total; i++) {
		pages.push(i)
	}
	const handleClik = page => {
		onClick(page)
	}
	return (
		<ul className="pagination pagination-sm">
			{pages.map((page, idx) => (
				<li
					className={`page-item ${
						current === idx + 1 ? 'active' : ''
					}`}
					key={page}
					onClick={handleClik.bind(this, page)}
				>
					<a className="page-link" href="#">
						{page}
					</a>
				</li>
			))}
		</ul>
	)
}

Pagination.propTypes = {
	current: PropTypes.number,
	total: PropTypes.number,
	onClick: PropTypes.func
}
