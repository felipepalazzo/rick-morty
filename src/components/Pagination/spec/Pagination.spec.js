import React from 'react'
import { shallow } from 'enzyme'
import Pagination from '../Pagination'

describe('Pagination', () => {
	const props = { current: 5, total: 10, onClick: jest.fn() }
	const wrapper = shallow(<Pagination {...props} />)
	it('renders without crashing', () => {
		expect(wrapper).toMatchSnapshot()
	})
	it('should render the correct num of pages', () => {
		expect(wrapper.find('li')).toHaveLength(props.total)
	})
	it('should accept a onClick function', () => {
		wrapper
			.find('li')
			.first()
			.prop('onClick')()
		expect(props.onClick).toHaveBeenCalledTimes(1)
	})
})
