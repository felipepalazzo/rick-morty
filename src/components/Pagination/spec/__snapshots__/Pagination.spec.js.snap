// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders without crashing 1`] = `ShallowWrapper {}`;
